src/nidigital/templates/session.py/fancy_fetch_capture_waveform.py.mako

```diff
@@ -52,11 +53,11 @@
             start = i * actual_samples_per_waveform
             end = start + actual_samples_per_waveform
             if sys.version_info.major >= 3:
-                measurements.append(Measurement(data=mv[start:end], site=site_list[i]))
+                waveforms.append(Waveform(site=site_list[i], data=mv[start:end]))
             else:
                 # memoryview in Python 2 doesn't support numeric types, so we copy into an array.array to put in the wfm. :( You should be using Python 3!
                 # Or use the _into version. memoryview in Python 2 only supports string and bytearray, not array.array or numpy.ndarray of arbitrary types.
-                measurements.append(Measurement(data=array.array('d', data[start:end]), site=site_list[i]))
+                waveforms.append(Waveform(site=site_list[i], data=array.array('L', data[start:end])))
 
-        return measurements
+        return waveforms
```

Comment:
We are missing test coverage for this hand-coded function. Seems like we could write a unit test by mocking  the private function (if we can get it to be code-generated) else we could mock the library.

Or can we get adequate coverage through sytem tests?

Reply:
None of the main drivers currently have unit tests and none do any mocking.

This would be a large change to how we are doing things.

Reply:
Or can we get adequate coverage through sytem tests?